sourced/ml/transformers/basic.py

```diff
@@ -259,12 +259,12 @@ class UastExtractor(Transformer):
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
 
-    def __call__(self, files: Union[BlobsDataFrame, BlobsWithLanguageDataFrame]) -> DataFrame:
+    def __call__(self, files: Union[BlobsDataFrame, BlobsWithLanguageDataFrame]) -> RDD:
```

Comment:
Nah, not so easy :) Refer to https://github.com/src-d/vecino/blob/master/vecino/repo2bow.py#L33

Reply:
Can't I factorize the Moder, and put in https://github.com/src-d/ml/master/sourced/ml/transformers/basic.py#L375 ?

Reply:
The idea is to split the pipeline into as many separate pieces as possible so that we reach the true specialization of the classes. So no.